src/itensornetworkfunction.jl

```diff
@@ -73,6 +80,31 @@ function project(fitn::ITensorNetworkFunction, ind_to_ind_value_map)
   return fitn
 end
 
+" Naive integration of a function in all dimensions ∫₀¹f({r})d{r} "
+function integrate(fitn::ITensorNetworkFunction; alg=default_contraction_alg(), kwargs...)
+  fitn = copy(fitn)
+  s = indsnetwork(indsnetworkmap(fitn))
+  for v in vertices(fitn)
+    indices = inds(s, v)
+    fitn[v] = fitn[v] * ITensor(eltype(fitn[v]), 0.5, indices...)
+  end
+  return scalar(itensornetwork(fitn); alg, kwargs...)
+end
+
+function integrate(
+  O::TreeTensorNetwork,
```

Comment:
We can drop the `TreeTensorNetwork` and make `O` an `AbstractITensorNetwork` here. The version of `inner` being called here will work regardless

Reply:
Maybe also call `O`, `operator` as that is clearer what its action is

Reply:
I'll rename the operator function then I think, so we don't overload names too much